data-prepper-core/src/main/java/org/opensearch/dataprepper/core/peerforwarder/PeerForwardingProcessorDecorator.java

```diff
@@ -24,12 +25,14 @@
 public class PeerForwardingProcessorDecorator implements Processor<Record<Event>, Record<Event>> {
     private final PeerForwarder peerForwarder;
     private final Processor innerProcessor;
+    private final boolean peerForwardingDisabled;
 
     public static List<Processor> decorateProcessors(
             final List<Processor> processors,
             final PeerForwarderProvider peerForwarderProvider,
             final String pipelineName,
             final String pluginId,
+            final List<String> noPeerForwardingKeys,
```

Comment:
Let's take this in as a `Collection<Collection<String>>` to indicate that the order does not matter. In fact we use it as a set.

Also, note my changes above for why it is a nested collection.